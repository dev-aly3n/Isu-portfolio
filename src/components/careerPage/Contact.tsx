interface props {
  contact: { site: string; username: string; logo: string; link: string };
}

const Contact: React.FC<props> = ({ contact }) => {
  return (
    <div className="contact-container">
      <a href={contact.link} target="_blank">
        <img src={`/media/logos/${contact.logo}.jpg`} />
        <div>
          <h4>{contact.site}</h4>
          <p>{contact.username}</p>
        </div>
      </a>
    </div>
  );
};

export default Contact;
