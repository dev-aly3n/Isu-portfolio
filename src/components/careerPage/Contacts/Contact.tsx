import Image from "next/image";

interface props {
  contact: { site: string; username: string; logo: string; link: string };
}

const Contact: React.FC<props> = ({ contact }) => {
  return (
    <div className="contact-container">
      <a href={contact.link} target="_blank">
        <span>
          <Image
            alt={contact.site}
            src={`/media/logos/${contact.logo}.jpg`}
            layout="fill"
            objectFit="cover"
          />
        </span>
        <div>
          <h4>{contact.site}</h4>
          <p>{contact.username}</p>
        </div>
      </a>
    </div>
  );
};

export default Contact;
