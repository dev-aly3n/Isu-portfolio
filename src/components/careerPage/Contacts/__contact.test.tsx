import { render, screen } from "@testing-library/react";
import React, { useRef } from "react";
import { RefObject } from "react";
import Contact from "./Contact";

const contactData = {
    site: "google",
    username: "goleGole",
    logo:"/media/logos/instagram.jpg",
    link:"https://instagram.com"
}

describe("career/contacts", () => {
  it("snap", () => {
    const { container } = render(<Contact contact={contactData} />);
    expect(container).toMatchSnapshot();
  });
});
