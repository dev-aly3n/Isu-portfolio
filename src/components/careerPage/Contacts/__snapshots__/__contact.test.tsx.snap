// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`career/contacts snap 1`] = `
<div>
  <div
    class="contact-container"
  >
    <a
      href="https://instagram.com"
      target="_blank"
    >
      <span>
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="google"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </span>
      <div>
        <h4>
          google
        </h4>
        <p>
          goleGole
        </p>
      </div>
    </a>
  </div>
</div>
`;
